widgetssdk/src/main/java/com/glia/widgets/view/snackbar/SnackBarDelegate.kt

```diff
@@ -0,0 +1,109 @@
+package com.glia.widgets.view.snackbar
+
+import android.app.Activity
+import android.view.View
+import android.view.ViewGroup.MarginLayoutParams
+import androidx.annotation.ColorRes
+import androidx.annotation.IdRes
+import androidx.annotation.VisibleForTesting
+import androidx.core.view.ViewCompat
+import androidx.core.view.WindowInsetsCompat
+import androidx.core.view.updateLayoutParams
+import androidx.core.view.updateMargins
+import com.glia.widgets.Constants
+import com.glia.widgets.R
+import com.glia.widgets.StringProvider
+import com.glia.widgets.call.CallActivity
+import com.glia.widgets.chat.ChatActivity
+import com.glia.widgets.helper.getColorCompat
+import com.glia.widgets.helper.insetsController
+import com.glia.widgets.helper.rootView
+import com.glia.widgets.helper.rootWindowInsetsCompat
+import com.glia.widgets.view.unifiedui.theme.SnackBarTheme
+import com.glia.widgets.view.unifiedui.theme.UnifiedTheme
+import com.google.android.material.snackbar.Snackbar
+
+internal abstract class SnackBarDelegate(
+    view: View, stringProvider: StringProvider, snackBarTheme: SnackBarTheme?
+) {
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
+    @get:IdRes
+    open val anchorViewId: Int? = null
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
+    open val marginBottom: Int? = null
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
+    @get:ColorRes
+    open val fallbackBackgroundColor: Int = R.color.glia_base_dark_color
+
+    @VisibleForTesting(otherwise = VisibleForTesting.PROTECTED)
+    @get:ColorRes
+    open val fallbackTextColor: Int = R.color.glia_base_light_color
+
+    @VisibleForTesting
+    val snackBar: Snackbar by lazy { makeSnackBar(view, stringProvider, snackBarTheme) }
+
+    fun show() = snackBar.show()
+
+    private fun makeSnackBar(view: View, stringProvider: StringProvider, snackBarTheme: SnackBarTheme?): Snackbar {
+        val bgColor = snackBarTheme?.backgroundColorTheme?.primaryColor ?: view.getColorCompat(fallbackBackgroundColor)
+        val textColor = snackBarTheme?.textColorTheme?.primaryColor ?: view.getColorCompat(fallbackTextColor)
+        val message = stringProvider.getRemoteString(R.string.live_observation_indicator_message)
```

Comment:
I think you should eather change the method name to something like `makeLiveObservationSnackBar` or extract the message as an argument e.g: `@StringRes messageId: Int`